pkg/install/error_test.go

```diff
@@ -0,0 +1,245 @@
+package install
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"errors"
+	"net/http"
+	"reflect"
+	"testing"
+
+	"github.com/Azure/go-autorest/autorest"
+	"github.com/Azure/go-autorest/autorest/azure"
+	"github.com/golang/mock/gomock"
+	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	mock_resources "github.com/Azure/ARO-RP/pkg/util/mocks/azureclient/mgmt/resources"
+)
+
+func TestIsResourceQuotaExceededError(t *testing.T) {
+	for _, tt := range []struct {
+		name                   string
+		isDetailedError        bool
+		originalIsOtherError   bool
+		originalIsServiceError bool
+		code                   interface{}
+		message                interface{}
+		wantBool               bool
+		wantMessage            string
+	}{
+		{
+			name: "Is not a DetailedError",
+		},
+		{
+			name:            "Does not contain original error",
+			isDetailedError: true,
+		},
+		{
+			name:                 "Original is not a ServiceError",
+			isDetailedError:      true,
+			originalIsOtherError: true,
+		},
+		{
+			name:                   "No code in ServiceError",
+			isDetailedError:        true,
+			originalIsServiceError: true,
+		},
+		{
+			name:                   "Quota exceeded but no message in ServiceError",
+			isDetailedError:        true,
+			originalIsServiceError: true,
+			code:                   "QuotaExceeded",
+		},
+		{
+			name:                   "Different code in ServiceError",
+			isDetailedError:        true,
+			originalIsServiceError: true,
+			code:                   "AnotherCode",
+			message:                "Something happened",
+		},
+		{
+			name:                   "Quota exceeded",
+			isDetailedError:        true,
+			originalIsServiceError: true,
+			code:                   "QuotaExceeded",
+			message:                "Quota exceeded",
+			wantBool:               true,
+			wantMessage:            "Quota exceeded",
+		},
+	} {
+		t.Run(tt.name, func(t *testing.T) {
+			var err error
+			if tt.isDetailedError {
+				var originalErr error
+				if tt.originalIsServiceError {
+					originalErr = &azure.ServiceError{
+						Details: []map[string]interface{}{
+							{
+								"code":    tt.code,
+								"message": tt.message,
+							},
+						},
+					}
+				} else if tt.originalIsOtherError {
+					originalErr = errors.New("☹")
+				}
+				err = autorest.NewErrorWithError(originalErr, "", "", nil, "")
+			} else {
+				err = errors.New("☹")
+			}
+			isQuota, errMsg := isResourceQuotaExceededError(err)
+			if isQuota != tt.wantBool || errMsg != tt.wantMessage {
+				t.Error(tt.wantBool, tt.wantMessage)
+			}
+		})
+	}
+}
+
+func TestIsDeploymentActiveError(t *testing.T) {
+	for _, tt := range []struct {
+		name                    string
+		isDetailedError         bool
```

Comment:
I think it makes more sense to define an error as a field here (something like `inputError error`) and in cases do something like:

```
{
			name:            "Does not contain original error",
			inputError:     errors.New("some fake error"),
		},
```

So you will get rid of a lot of `if`s in the test body.

Reply:
I see your point. I was worried that there would be a lot of boilerplate which would be hard to read, but the nested conditionals I have may be as hard or even harder to read. I'll change it to what you suggested.

Reply:
Fixed